app/scripts/controllers/metamask-notifications/metamask-notifications.ts

```diff
@@ -729,6 +989,21 @@ export class MetamaskNotificationsController extends BaseController<
 
       const readIds = this.state.metamaskNotificationsReadList;
 
+      // For each rawOnChainNotifications check if the notification unread is false
+      // update the state metamaskNotificationsReadList removing the id from the list
+      rawOnChainNotifications.forEach(
+        (notification: OnChainRawNotification) => {
+          if (notification.unread === false) {
+            this.update((s) => {
+              s.metamaskNotificationsReadList =
+                s.metamaskNotificationsReadList.filter(
+                  (id) => id !== notification.id,
+                );
+            });
+          }
+        },
+      );
```

Comment:
Hmm sure we can add read state for onchain notifications... realistically on-chain notifications are already sent to server if they are marked as read.

Is this needed?

Reply:
if it works, it works. We can tidy in future.

Reply:
Ticket: https://consensyssoftware.atlassian.net/browse/NOTIFY-655

Reply:
Ok, I've already reviewed this part, and indeed it's an unnecessary piece of code. Please check the code, and if that's the case, let's close the ticket :)